sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentHealthcareEntities.cs

```diff
@@ -0,0 +1,117 @@
+
+namespace Azure.AI.TextAnalytics
+{
+    public partial class DocumentHealthcareEntities
+    {
+        /// <summary> Initializes a new instance of DocumentHealthcareEntities. </summary>
+        /// <param name="id"> Unique, non-empty document identifier. </param>
+        /// <param name="entities"> Healthcare entities. </param>
+        /// <param name="relations"> Healthcare entity relations. </param>
+        /// <param name="warnings"> Warnings encountered while processing document. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="id"/>, <paramref name="entities"/>, <paramref name="relations"/>, or <paramref name="warnings"/> is null. </exception>
+        internal DocumentHealthcareEntities(string id, IEnumerable<HealthcareEntity> entities, IEnumerable<HealthcareRelationInternal> relations, IEnumerable<TextAnalyticsWarningInternal> warnings)
+        {
+            if (id == null)
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
+            if (entities == null)
+            {
+                throw new ArgumentNullException(nameof(entities));
+            }
+            if (relations == null)
+            {
+                throw new ArgumentNullException(nameof(relations));
+            }
+            if (warnings == null)
+            {
+                throw new ArgumentNullException(nameof(warnings));
+            }
+
+            Id = id;
+            Entities = entities.ToList();
+            Relations = ResolveHealthcareRelations(entities, relations);
+            Warnings = Transforms.ConvertToWarnings(warnings.ToList());
+        }
+
+        internal DocumentHealthcareEntities(DocumentHealthcareEntitiesInternal documentHealthcareEntities)
+        {
+            Entities = documentHealthcareEntities.Entities;
+            Relations = ResolveHealthcareRelations(documentHealthcareEntities.Entities, documentHealthcareEntities.Relations);
+            Id = documentHealthcareEntities.Id;
+            Warnings = documentHealthcareEntities.Warnings != null ? Transforms.ConvertToWarnings(documentHealthcareEntities.Warnings) : null;
+        }
+
+        /// <summary> Initializes a new instance of DocumentHealthcareEntities. </summary>
+        /// <param name="id"> Unique, non-empty document identifier. </param>
+        /// <param name="entities"> Healthcare entities. </param>
+        /// <param name="relations"> Healthcare entity relations. </param>
+        /// <param name="warnings"> Warnings encountered while processing document. </param>
+        /// <param name="statistics"> if showStats=true was specified in the request this field will contain information about the document payload. </param>
+        internal DocumentHealthcareEntities(string id, IReadOnlyList<HealthcareEntity> entities, IReadOnlyList<HealthcareRelationInternal> relations, IReadOnlyList<TextAnalyticsWarningInternal> warnings, TextDocumentStatistics? statistics)
```

Comment:
could you point me to who uses each constructor? not sure I understand the need of 3

Reply:
I am using only 1 out of the 3, will remove the other 2.